mixins/provider/provider-mixin.js

```diff
@@ -1,9 +1,20 @@
+export class ProviderDelegate {
+	constructor(delegate) {
+		this.delegate = delegate;
+	}
+}
+
 export function provideInstance(node, key, obj) {
 	if (!node._providerInstances) {
 		node._providerInstances = new Map();
 		node.addEventListener('d2l-request-instance', e => {
 			if (node._providerInstances.has(e.detail.key)) {
-				e.detail.instance = node._providerInstances.get(e.detail.key);
+				const instance = node._providerInstances.get(e.detail.key);
+				if (instance instanceof ProviderDelegate) {
```

Comment:
I initially implemented this to just check for `Function`, but changed to accept `ProviderDelegate` because there a couple of existing usages of `provideInstance` that already specify a function. I think it is safer to accept this specific class, leaving the option to consumers to provide an actual function to requestors that doesn't get invoked by us here.

Reply:
Yeah I was thinking the same... in retrospect this should have just always taken a function (which could have in turn returned a function), but we're kind of stuck with this now.